crates/solidity/outputs/cargo/tests/src/doc_examples/using_queries.rs

```diff
@@ -0,0 +1,176 @@
+use std::path::Path;
+
+use anyhow::Result;
+use infra_utils::paths::PathExtensions;
+use semver::Version;
+use slang_solidity::kinds::RuleKind;
+use slang_solidity::language::Language;
+use slang_solidity::parse_output::ParseOutput;
+use slang_solidity::query::Query;
+
+fn parse_doc_input_file<T: AsRef<Path>>(path: T) -> Result<ParseOutput> {
+    let input_path =
+        Path::repo_path(Path::new("documentation/public/user-guide/inputs").join(path.as_ref()));
+
+    let input_path = input_path.unwrap_str();
+
+    let source = std::fs::read_to_string(input_path)?;
+    let source = source.trim();
+
+    let language = Language::new(Version::new(0, 8, 0))?;
+
+    Ok(language.parse(RuleKind::SourceUnit, source))
+}
+
+#[test]
+fn using_queries() -> Result<()> {
+    #[allow(unused_variables, clippy::items_after_statements)]
+    {
+        let parse_output = parse_doc_input_file("using-the-cursor.sol")?;
+        // --8<-- [start:creating-a-query]
+        use slang_solidity::query::Query;
+
+        // Any `Cursor` can be used to create a query.
+        let cursor = parse_output.create_tree_cursor();
+
+        let query = Query::parse("[ContractDefinition]").unwrap();
+        let result /* : QueryResultIterator */ = cursor.query(vec![query]);
+        // --8<-- [end:creating-a-query]
+    }
+
+    {
+        let parse_output = parse_doc_input_file("using-the-cursor.sol")?;
+        let cursor = parse_output.create_tree_cursor();
+        // --8<-- [start:listing-contract-names]
+        let mut names = vec![];
+
+        let query = Query::parse("[ContractDefinition ... @name [Identifier] ...]").unwrap();
+
+        for result in cursor.query(vec![query]) {
+            let bindings = result.bindings;
+            let cursors = bindings.get("name").unwrap();
+
+            let cursor = cursors.first().unwrap();
+
+            names.push(cursor.node().unparse());
```

Comment:
for getting contract names (and below for struct/enum names), what do you think about using label selectors (`name: _`) instead of the kind (`Identifier`)?
While it wont change the result in these specific examples, it is more accurate/strict in general, and demonstrates the effectiveness of labels.

Reply:
As long as we demonstrate all the options

Reply:
To keep the first example minimal, what do you think of doing something simple like just selecting the full range of the contract (just matching on the `ContractDefinition` kind)?

In the second example, we can then introduce labels to pick the contract/struct/enum/etc... names.

Reply:
I've adapted the example to match the entire range and print it, so the query here should be very minimal `@contract [ContractDefinition]`. I could also do the `[ContractDefinition]` but then there's not much info we can get because we don't have bindings, so the could only count the visits; that's not as exciting or helpful, so I've decided to do the binding, instead.

Reply:
Sounds great.